### ISSUE TITLE
Adding Roles to users logging in with OAuth providers

### ISSUE THREAD

**Issue by saucepleez** [Requarks/wiki#650]
Hello,

I would like to add roles to a user that has been authorized by Github login, but the name appears to be missing.

### Actual behavior
![image](https://user-images.githubusercontent.com/5113950/46591248-67764500-ca87-11e8-8d0a-de6894873ce2.png)
User is created after signing in with a GitHub account but is missing the user name.
### Expected behavior
Expected that the user would have a name assigned.
![image](https://user-images.githubusercontent.com/5113950/46591382-5e39a800-ca88-11e8-8475-e3dbfe8175eb.png)

### Steps to reproduce the behavior
- Vanilla install for WikiJS
- Create GitHub Secret/ID
- Update the config.yml
- Login with Github Account
- Logout, then login as admin

Thanks.

**Comment by saucepleez** [comment]
Hello,

I figured out what is causing the issue...  When a user logs in from GitHub for the first time, the name is either null or empty when the record is created:

![image](https://user-images.githubusercontent.com/5113950/46819868-cfac7b80-cd52-11e8-8611-7af4222a652f.png)

Once the user assigns a name to themselves then they show up for administration:

![image](https://user-images.githubusercontent.com/5113950/46819982-18fccb00-cd53-11e8-9055-5b0b3cc6137c.png)

But the user may never assign themselves a name, but they should always be able to be handled through the administration panel.

Suggestions for the change:
- When a user logs in from OAuth, create a standard name "Default User", "User1023023", etc... I hesitate to use emails as usernames due to spam/email name sensitivity concerns...

- Administration Panel should detect if the name is null or empty and populate a wrapped link.  A link should always be populated.
`
<a href="/admin/users/5bbacaa881c0e820d7ccdc0d">Default User/Not Defined</a>
`

What do you think? Are you still accepting pull requests for 1.X?

**Comment by NGPixel** [comment]
Yes PRs are still accepted for 1.x as long as they are bug fixes, not new features or introducing breaking changes.

**Comment by constructive-realities** [comment]
Is there a work around for this?   

can i delete a github social user somehow and setup it up correctly?
